lib/Sema/CSSimplify.cpp

```diff
@@ -3672,7 +3675,9 @@ ConstraintSystem::simplifyRestrictedConstraint(ConversionRestrictionKind restric
                                         getConstraintLocator(locator));
         
         Constraint *disjunctionChoices[] = {int8Con, uint8Con, voidCon};
-        addDisjunctionConstraint(disjunctionChoices, locator);
+        addUnsolvedConstraint(
+          Constraint::createDisjunction(*this, disjunctionChoices,
+                                        getConstraintLocator(locator)));
```

Comment:
I might be missing something about what you're doing here, but unlike the other cases where I think we're tending to reform the same constraint, in this case we're adding a new / different constraint. The existing code is never going to attempt to simplify this, so perhaps it's appropriate, but it looks a little odd offhand.